### PR TITLE
#3441 Implement plugins page to add, delete & upload plugin

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3441 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import Card from '@material-ui/core/Card';
+import CardActionArea from '@material-ui/core/CardActionArea';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+
+import { useAddPluginDialog } from 'context';
+import PluginSelector from './PluginSelector';
+import { Wrapper } from './AddPluginCardStyles';
+
+AddPluginCard.propTypes = {};
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js]: Redundant :)

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+function AddPluginCard() {
+  const { open } = useAddPluginDialog();
+  return (
+    <Wrapper>
+      <CardActionArea
+        onClick={() => {
+          open();
+        }}
+      >
+        <Card className="">
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js]: Is this intentional?

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { Wrapper } from './AddPluginCardStyles';
+
+AddPluginCard.propTypes = {};
+
+function AddPluginCard() {
+  const { open } = useAddPluginDialog();
+  return (
+    <Wrapper>
+      <CardActionArea
+        onClick={() => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js]: Could be changed to `onClick={open}` :)

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/PluginCardStyles.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        transform: rotate(180deg);
+      }
+    }
+    .services-detail {
+      .MuiChip-root {
+        height: 20px;
+        border-radius: 4px;
+      }
+      .service-name:hover {
+        background-color: rgba(0, 0, 0, 0.08);
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/Plugin/PluginCardStyles.js]: There's a have state for the color :)

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/PluginSelector.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+import { FileUpload } from '../File';
+import {
+  StyledDialogTitle,
+  StyledDialogContent,
+  StyledDialogActions,
+  FileFilter,
+  FileList,
+} from './PluginSelectorStyles';
+
+PluginSelector.propTypes = {};
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Edit/Plugin/PluginSelector.js]: Same here

File: ohara-manager/client/src/context/api/utils.js [hunk 6]
```diff
@@ -26,42 +26,23 @@ export const validate = values => {
...
+export const generateClusterResponse = params => {
+  const { values = {}, stageValues = {}, inspectInfo = {} } = params;
 
-  if (!isEmpty(inspectInfo.imageName)) result.imageName = inspectInfo.imageName;
-  if (!isEmpty(inspectInfo.classInfos))
-    result.classInfos = inspectInfo.classInfos;
+  if (isEmpty(inspectInfo)) {
+    return {
+      ...values,
+      // Flatten settings, remove them when appropriate
```
Comment (konekoya) [inline ohara-manager/client/src/context/api/utils.js]: 👍

File: ohara-manager/client/src/context/index.js [hunk 7]
```diff
@@ -14,162 +14,19 @@
...
+export * from 'context/file/FileContext';
+export * from 'context/node/NodeContext';
+export * from 'context/pipeline/PipelineContext';
+export * from 'context/stream/StreamContext';
+export * from 'context/topic/TopicContext';
+export * from 'context/worker/WorkerContext';
+export * from 'context/workspace/WorkspaceContext';
+export * from 'context/workspace/EditWorkspaceContext';
+export * from 'context/zookeeper/ZookeeperContext';
+export * from 'context/SnackbarContext';
```
Comment (konekoya) [inline ohara-manager/client/src/context/index.js]: 👍

File: ohara-manager/client/src/theme/muiTheme.js [hunk 8]
```diff
@@ -76,9 +92,9 @@ const MuiTheme = createMuiTheme({
       contrastText: '#FFFFFF',
     },
     warning: {
-      main: '#FFAA00',
-      light: '#FFDC4A',
-      dark: '#C67B00',
+      main: '#ffc107',
+      light: '#ffd54f',
+      dark: '#ffa000',
```
Comment (konekoya) [inline ohara-manager/client/src/theme/muiTheme.js]: Thanks :)

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js]: Is my careless :(

File: ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/components/Workspace/Edit/Plugin/AddPluginCard.js]: Thanks for reminding :)

**Pull request merged**